storagemarket/types.go

```diff
@@ -27,43 +27,56 @@ type Balance struct {
 	Available tokenamount.TokenAmount
 }
 
-type DealState = uint64
+type StorageDealStatus = uint64
 
 const (
-	DealUnknown  = DealState(iota)
-	DealRejected // Provider didn't like the proposal
-	DealAccepted // Proposal accepted
-	DealStaged   // Data put into the sector
-	DealSealing  // Data in process of being sealed
-
-	DealFailed
-	DealComplete
+	StorageDealUnknown = StorageDealStatus(iota)
+	StorageDealProposalNotFound
+	StorageDealProposalRejected
+	StorageDealProposalAccepted
+	StorageDealStaged
+	StorageDealSealing
+	StorageDealProposalSigned
+	StorageDealPublished
+	StorageDealCommitted
+	StorageDealActive
+	StorageDealFailing
+	StorageDealRecovering
+	StorageDealExpired
+	StorageDealNotFound
 
 	// Internal
```

Comment:
^^ NB: These are all "internal" in the sense that they are not actually states that exist outside these markets. In any case, I don't see much difference.